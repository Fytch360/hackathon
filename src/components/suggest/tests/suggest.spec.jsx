import React from 'react'
import { shallow } from 'enzyme'

import { Suggest } from '..'

describe('<Suggest />', () => {

    test('Is available', () => {
        expect(Suggest).toBeDefined()
    })

    test('Snapshot', () => {
        const onClickMock = jest.fn()
        const optionsValue = ['test0', 'test1', 'test2', 'test3', 'test4']

        const wrapper = shallow(<Suggest options={optionsValue} onClick={onClickMock} size="md" />)

        expect(wrapper).toMatchSnapshot()
    })

    test('Should render option text in every suggest', () => {
        const optionsValue = ['test0', 'test1', 'test2', 'test3', 'test4']
        const wrapper = shallow(<Suggest options={optionsValue} />)

        wrapper.find('SuggestButtonStyled').map((child, index) => expect(child.text()).toBe(optionsValue[index]))
    })

    test('Should call onClick only once for click', () => {
        const onClickMock = jest.fn()
        const optionsValue = ['test0', 'test1', 'test2', 'test3', 'test4']

        const wrapper = shallow(<Suggest options={optionsValue} onClick={onClickMock} />)

        expect(onClickMock.mock.calls.length).toBe(0)

        wrapper.find('SuggestButtonStyled').first()
            .simulate('click', {})

        expect(onClickMock.mock.calls.length).toBe(1)
    })

    test('Should return correct option value when suggest was clicked', () => {
        const onClickMock = jest.fn()
        const optionsValue = ['test0', 'test1', 'test2', 'test3', 'test4']

        const wrapper = shallow(<Suggest options={optionsValue} onClick={onClickMock} />)

        wrapper.find('SuggestButtonStyled').at(2)
            .simulate('click', {})

        expect(onClickMock.mock.calls[0][0]).toBe(optionsValue[2])

        wrapper.find('SuggestButtonStyled').at(4)
            .simulate('click', {})

        expect(onClickMock.mock.calls[1][0]).toBe(optionsValue[4])
    })

    test('Should\'nt call onClick if suggest is disabled', () => {
        const onClickMock = jest.fn()
        const optionsValue = ['test0', 'test1', 'test2', 'test3', 'test4']

        const wrapper = shallow(<Suggest options={optionsValue} onClick={onClickMock} disabled />)

        expect(onClickMock.mock.calls.length).toBe(0)

        wrapper.children().first()
            .simulate('click', {})

        expect(onClickMock.mock.calls.length).toBe(0)
    })
})
