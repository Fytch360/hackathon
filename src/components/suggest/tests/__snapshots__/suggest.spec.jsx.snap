// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Suggest /> Snapshot 1`] = `
<MarginWrapper
  size="md"
  verticalMargin="micro"
>
  <SuggestsListStyled
    role="listbox"
  >
    <SuggestWrapperStyled
      key="test0"
      role="presentation"
    >
      <SuggestButtonStyled
        aria-label=" test0"
        disabled={false}
        onClick={[Function]}
        size="md"
        type="button"
      >
        <TypographyStyled
          horizontalPadding="nano"
          size="md"
          verticalPadding="nano"
        >
          test0
        </TypographyStyled>
      </SuggestButtonStyled>
    </SuggestWrapperStyled>
    <SuggestWrapperStyled
      key="test1"
      role="presentation"
    >
      <SuggestButtonStyled
        aria-label=" test1"
        disabled={false}
        onClick={[Function]}
        size="md"
        type="button"
      >
        <TypographyStyled
          horizontalPadding="nano"
          size="md"
          verticalPadding="nano"
        >
          test1
        </TypographyStyled>
      </SuggestButtonStyled>
    </SuggestWrapperStyled>
    <SuggestWrapperStyled
      key="test2"
      role="presentation"
    >
      <SuggestButtonStyled
        aria-label=" test2"
        disabled={false}
        onClick={[Function]}
        size="md"
        type="button"
      >
        <TypographyStyled
          horizontalPadding="nano"
          size="md"
          verticalPadding="nano"
        >
          test2
        </TypographyStyled>
      </SuggestButtonStyled>
    </SuggestWrapperStyled>
    <SuggestWrapperStyled
      key="test3"
      role="presentation"
    >
      <SuggestButtonStyled
        aria-label=" test3"
        disabled={false}
        onClick={[Function]}
        size="md"
        type="button"
      >
        <TypographyStyled
          horizontalPadding="nano"
          size="md"
          verticalPadding="nano"
        >
          test3
        </TypographyStyled>
      </SuggestButtonStyled>
    </SuggestWrapperStyled>
    <SuggestWrapperStyled
      key="test4"
      role="presentation"
    >
      <SuggestButtonStyled
        aria-label=" test4"
        disabled={false}
        onClick={[Function]}
        size="md"
        type="button"
      >
        <TypographyStyled
          horizontalPadding="nano"
          size="md"
          verticalPadding="nano"
        >
          test4
        </TypographyStyled>
      </SuggestButtonStyled>
    </SuggestWrapperStyled>
  </SuggestsListStyled>
</MarginWrapper>
`;
